snapcraft/plugins/kernel.py

```diff
@@ -142,16 +141,14 @@ def __init__(self, name, options):
         self.make_install_targets.extend(self._get_fw_install_targets())
 
         self.os_snap = os.path.join(self.sourcedir, 'os.snap')
-        self._target_arch = get_machine_info(target_machine)
 
-    def set_target_machine(self, machine):
-        self._target_arch = get_machine_info(machine)
+    def set_target_machine(self):
```

Comment:
A "set" function with no parameter seems odd. Is this something that can be done in the initializer now?